lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -84,8 +84,16 @@ def schedule
       end
 
       # reloads the schedule from redis
-      def reload_schedule!
+      def reload_schedule!(verify_with: nil, max_retries: 7)
+        retries = 0
+
         @schedule = all_schedules
+        while verify_with && retries < max_retries &&
+              !(@schedule && @schedule.eql?(verify_with))
+          @schedule = all_schedules
+          retries += 1
+          sleep 2**(retries - rand + 1) * 0.1 unless @schedule
+        end
```

Comment:
I think I would rewrite this method a bit for clarity.. And I don't like `while` loops too much :-)
```ruby
def reload_schedule(verify_with: nil, max_retries: 7)
  (1..max_retries).each do |tries|
    schedule = all_schedules

    if schedule.eql?(verify_with)
      return @schedule = schedule
    end

    sleep 0.1 * (2**tries)
  end
end
```